fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/FixityServiceImpl.java

```diff
@@ -49,8 +73,37 @@
     }
 
     @Override
-    public Collection<URI> checkFixity(final Binary binary, final Collection<String> algorithms)
-            throws UnsupportedAlgorithmException {
-        return null;
+    public RdfStream checkFixity(final Binary binary)
+            throws InvalidChecksumException {
+        final Model model = createDefaultModel();
+        final Resource subject = model.createResource(binary.getId());
+        final Resource fixityResult = model.createResource(
+                binary.getFedoraId().resolve("#" + UUID.randomUUID().toString()).getFullId());
+        model.add(subject, HAS_FIXITY_RESULT, fixityResult);
+        model.add(fixityResult, type, PREMIS_FIXITY);
+        model.add(fixityResult, type, PREMIS_EVENT_OUTCOME_DETAIL);
+        model.add(fixityResult, HAS_SIZE, createTypedLiteral(binary.getContentSize()));
+
+        // Built for more than one digest in anticipation of FCREPO-3419
+        final List<URI> existingDigestList = new ArrayList<>();
+        existingDigestList.add(binary.getContentDigest());
+
+        final var digestAlgs = existingDigestList.stream()
+                .map(URI::toString)
+                .map(a -> a.replace("urn:", "").split(":")[0])
+                .map(DIGEST_ALGORITHM::fromAlgorithm)
+                .collect(Collectors.toList());
+
+        final MultiDigestInputStreamWrapper digestWrapper = new MultiDigestInputStreamWrapper(
+                binary.getContent(), existingDigestList, digestAlgs);
```

Comment:
Haven't looked at this PR in depth. Was just browsing and noticed this. `binary.getContent()` should be called in a try-with block so that it is closed.